pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -66,14 +66,18 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(env *model.Environme
 				break
 			}
 		}
+		listenerPort := int(portNumber)
+		if si != nil {
+			listenerPort = si.Endpoint.Port
+		}
```

Comment:
Since we're not using the ServiceInstance anywhere, why not declare listenerPort just above the for loop that finds the ServiceInstance and setting `listenerPort` when it finds it (instead of setting the `si` variable).

Reply:
Ah, nevermind. Didn't see that we're using `si` further down.

Reply:
Oh, but my comment is still valid. You could set `listenerPort` just after we set `si`.

Reply:
True. Will set it in the loop.